app/components/form/instance.js

```diff
@@ -111,9 +112,16 @@ export default class InstanceComponent extends Component {
     this.args.onCancel();
   }
 
-  async onInit() {
+  @action
+  async onRestore(historicalInstance) {
```

Comment:
Shouldn't restoring persist? Which it does not appear to do know.

Reply:
so indeed it doesn't but this was intentional as we don't have a preview/show changes feature yet. So the 'terugzetten' button currently only sets the values in the form but doesn't save. I have now tried to make this more obvious by only enabling the update button when the form has changes (which it has if you restore a previous version)

Also, you may have had the feeling that hitting the button didn't do anything because the current version is the top one, so you'd reset it to itself. i have removed the button in that case and marked is as the current version. 

Legacy data (not created by us) does not have an initial version (didn't go through the form content service. I will create another story to have a fallback for such cases, where the form content service creates a history item before updates if the current item doesn't have a history yet.